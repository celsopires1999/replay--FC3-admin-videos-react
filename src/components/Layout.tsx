import { Container } from "@mui/system";
import React from "react";

export function Layout({ children }: { children: React.ReactNode }) {
  return (
    <Container
      maxWidth="lg"
      sx={{
        mt: 4,
        mb: 4,
        color: "white",
        bgcolor: (theme) => theme.palette.grey[900],
      }}
    >
      {children}
    </Container>
  );
}
