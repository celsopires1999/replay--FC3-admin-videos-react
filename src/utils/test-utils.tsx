import type { PreloadedState } from "@reduxjs/toolkit";
import type { RenderOptions } from "@testing-library/react";
import { render } from "@testing-library/react";
import { SnackbarProvider } from "notistack";
import React, { PropsWithChildren } from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { AppStore, RootState, setupStore } from "../app/store";
import { ThemeProvider } from "@mui/system";
import { appTheme } from "../config/theme";

interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export function renderWithProviders(
  ui: React.ReactElement,
  { store = setupStore(), ...renderOptions }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return (
      <Provider store={store}>
        <ThemeProvider theme={appTheme}>
          <BrowserRouter>
            <SnackbarProvider>{children}</SnackbarProvider>
          </BrowserRouter>
        </ThemeProvider>
      </Provider>
    );
  }

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from "@testing-library/react";
