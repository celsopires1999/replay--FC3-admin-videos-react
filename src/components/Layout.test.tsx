// import { render } from "@testing-library/react";
import { renderWithProviders } from "../utils/test-utils";
import { Layout } from "./Layout";

describe("Layout Unit Tests", () => {
  it("should render layout correctly", () => {
    const { asFragment } = renderWithProviders(
      <Layout>
        <div>Test</div>
      </Layout>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
