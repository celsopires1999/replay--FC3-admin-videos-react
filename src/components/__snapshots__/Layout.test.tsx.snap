// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Unit Tests should render layout correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-mdbk7v-MuiContainer-root"
  >
    <div>
      Test
    </div>
  </div>
</DocumentFragment>
`;
